File: EntitiesCs/Account.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace EntitiesCs
{
    public abstract class Account
    {
        private int number;
        protected decimal balance;  // permite acceder a este campo en las subclases
		private Customer customer;

		public Account(Customer customer, int number, decimal balance)    // encadena  el constructor sin parámetros
        {
            Enabled = true;
            Number = number;
            this.balance = balance;
            Customer = customer;
		}
        public int Number
        {
            get
            {
                return number;
            }
            set
            {
                if (value < 10000)   // regla para aceptar el valor
                    throw new ArgumentException("Numero de cuenta debe ser 10000 o superior");
                number = value;
            }
        }
		public Customer Customer
		{
			get => customer;
            //internal set
			set
			{
                value.AddAccount(this);
				customer = value;
			}
		}
		public decimal Balance { get => balance; }
        public bool Enabled { get; set; }
        public abstract void Withdraw(decimal value); //Extracción
        public void Deposit(decimal value)  // metodo virtual permite ser sobreescrito
        {
            balance += value;
        }
    }
}

File: EntitiesVb/Account.vb
Public MustInherit Class Account
    Private _number As Integer
    Protected _balance As Decimal

    Public Sub New(customer As Customer, number As Integer, balance As Decimal)
        MyBase.New()
        _Enabled = True
        Me.Number = number
		_balance = balance
		Me.Customer = customer
	End Sub
    Public Property Number As Integer
        Get
            Return _number
        End Get
        Set(value As Integer)
            If value > 9999 Then
                _number = value
            End If
        End Set
    End Property
    Public ReadOnly Property Balance As Decimal
        Get
            Return _balance
        End Get
    End Property
    Private _customer As Customer
    Public Property Customer As Customer
        Get
            Return _customer
        End Get
		'Friend Set(value As Customer)
		Set(value As Customer)
			value.AddAccount(Me)
			_customer = value
        End Set
    End Property
    Public Property Enabled As Boolean
    Public Sub Deposit(value As Decimal)
        _balance += value
    End Sub

    Public MustOverride Sub Withdraw(value As Decimal)

    Public Overrides Function ToString() As String
        Return "Account " & Number
    End Function
End Class
